### PR TITLE
Bash completion and __init__.py lag

### PR DESCRIPTION
**Pull request by anselor**
For #369 
I've removed all of the default imports in `__init__.py` and updated all of the examples and unit tests to import directly from the new locations.
This is the first iteration of the bash completion capability. I don't know how to unit test the bash completion code - I'm open to any ideas there. When I have a bit more time I intent to detach this from argcomplete and implement our own bash bridge script that may be a better fit for what our implementation does.

### CODE REVIEW

**Comment by tleonhardt** [comment]
The only way I know how to unit test things like Bash tab-completion are using ``pexpect``.  I've used this successfully on my own machine and VMs, however I've never gotten it to work correctly on either of our CI systems (TravisCI or AppVeyor).

File: cmd2/__init__.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 #
 # -*- coding: utf-8 -*-
-#
-from .cmd2 import __version__, Cmd, set_posix_shlex, set_strip_quotes, AddSubmenu, CmdResult, categorize
-from .cmd2 import with_argument_list, with_argparser, with_argparser_and_unknown_args, with_category
+
+
```
Comment (tleonhardt) [inline cmd2/__init__.py]: Recommend configuring IDE/editor to strip trailing newlines after the last one

File: cmd2/argcomplete_bridge.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            """
+            self.__init__(argument_parser, always_complete_options=always_complete_options, exclude=exclude,
+                          validator=validator, print_suppressed=print_suppressed, append_space=append_space,
+                          default_completer=default_completer)
+
+            if "_ARGCOMPLETE" not in os.environ:
+                # not an argument completion invocation
+                return
+
+            global debug_stream
```
Comment (tleonhardt) [inline cmd2/argcomplete_bridge.py]: Global variable **debug_stream** isn't defined at the module level.  Should it be?

Also, is this really a constant?  If so maybe use all uppercase and put it at the module level?

File: cmd2/argcomplete_bridge.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            try:
+                debug_stream = os.fdopen(9, "w")
+            except:
```
Comment (tleonhardt) [inline cmd2/argcomplete_bridge.py]: Never use a bare **except** unless you are 100% sure that is what you really want to do (and it probably isn't).  Ideally, catch a specific tuple of exceptions.  But worst case, catch **Exception**:
```Python
except Exception:
    blah
```

This same comment applies a few lines below.

File: examples/alias_startup.py [hunk 4]
```diff
@@ -6,10 +6,10 @@
 """
 import argparse
```
Comment (tleonhardt) [inline examples/alias_startup.py]: I'm pretty sure that both ``argparse`` and ``pyparsing`` are unused imports within this example and should be removed

File: examples/alias_startup.py [hunk 5]
```diff
@@ -6,10 +6,10 @@
...
 
-import cmd2
+from cmd2 import cmd2
 import pyparsing
 
-from cmd2 import with_argument_list, with_argparser, with_argparser_and_unknown_args
+from cmd2.cmd2 import with_argument_list, with_argparser, with_argparser_and_unknown_args
```
Comment (tleonhardt) [inline examples/alias_startup.py]: Actually I think all of the imports in this file are unused either than the **from cmd2 import cmd2**

File: examples/help_categories.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 A sample application for tagging categories on commands.
 """
 
-from cmd2 import Cmd, categorize, __version__, with_argparser, with_category
+from cmd2.cmd2 import Cmd, categorize, __version__, with_argparser, with_category
 import argparse
```
Comment (tleonhardt) [inline examples/help_categories.py]: ``argparse`` is built-in to Python, so should generally be imported at the top before 3rd-party or custom imports.

File: cmd2/argcomplete_bridge.py (hunk 3 above)
Comment (anselor) [reply cmd2/argcomplete_bridge.py]: This is all code inherited from argcomplete. There are just a few places that I've changed to hijack their behavior.

File: cmd2/argcomplete_bridge.py (hunk 2 above)
Comment (anselor) [reply cmd2/argcomplete_bridge.py]: I probably need to import that from argcomplete.

File: cmd2/argcomplete_bridge.py (hunk 3 above)
Comment (tleonhardt) [reply cmd2/argcomplete_bridge.py]: Just because ``argcomplete`` did it in a crappy way doesn't mean you should continue to do so as well ;-)

Comment (anselor) [reply cmd2/argcomplete_bridge.py]: I haven't figured out how I want to do it yet, but I intend to borrow the bash completion script, install script, and other supporting scripts from argcomplete and completely drop any dependency we have on them. We're currently only using them to get access to their bash scripts.

Comment (tleonhardt) [reply cmd2/argcomplete_bridge.py]: I like the idea of not having even an optional dependency on ``argcomplete`` in the long run ...

File: cmd2/argcomplete_bridge.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+            try:
+                argcomplete.debug_stream = os.fdopen(9, "w")
+            except IOError:
+                argcomplete.debug_stream = sys.stderr
+
+            if output_stream is None:
+                try:
+                    output_stream = os.fdopen(8, "wb")
+                except IOError:
```
Comment (tleonhardt) [inline cmd2/argcomplete_bridge.py]: Thank you for fixing the bare excepts!

**Pull request merged**